mypy/meet.py

```diff
@@ -109,7 +115,10 @@ def visit_unbound_type(self, t: UnboundType) -> Type:
         if isinstance(self.s, Void) or isinstance(self.s, ErrorType):
             return ErrorType()
         elif isinstance(self.s, NoneTyp):
-            return self.s
+            if experiments.STRICT_OPTIONAL:
+                return AnyType()
+            else:
+                return self.s
         else:
             return AnyType()
 
```

Comment:
I suppose this should also have `elif isinstance(self.s, UninhabitedType): return self.s`, though I don't know specifically what this is for.